app/scripts/controllers/detect-tokens.js

```diff
@@ -24,36 +22,51 @@ export default class DetectTokensController {
     preferences,
     network,
     keyringMemStore,
+    tokenList,
   } = {}) {
     this.preferences = preferences;
     this.interval = interval;
     this.network = network;
     this.keyringMemStore = keyringMemStore;
+    this.tokenList = tokenList;
+  }
+
+  async _getTokenBalances(tokens) {
+    const ethContract = this.web3.eth
+      .contract(SINGLE_CALL_BALANCES_ABI)
+      .at(SINGLE_CALL_BALANCES_ADDRESS);
+    return new Promise((resolve, reject) => {
+      ethContract.balances([this.selectedAddress], tokens, (error, result) => {
+        if (error) {
+          return reject(error);
+        }
+        return resolve(result);
+      });
+    });
   }
 
   /**
-   * For each token in @metamask/contract-metadata, find check selectedAddress balance.
+   * For each token in tokenlist ptovided by the TokenListController, find check selectedAddress balance.
    */
   async detectNewTokens() {
     if (!this.isActive) {
       return;
     }
-    if (this._network.store.getState().provider.chainId !== MAINNET_CHAIN_ID) {
+    if (Object.keys(this._tokenList.state.tokenList).length <= 0) {
       return;
     }
 
     const tokensToDetect = [];
     this.web3.setProvider(this._network._provider);
-    for (const contractAddress in contracts) {
+    const apiTokens = this._tokenList.state.tokenList;
```

Comment:
I feel like this variable should just be `tokenList`. I think you're trying to call out that these tokens are coming from the tokenList API as opposed to our static list but I think this is more confusing?